### PR TITLE
ARROW-8092: [CI][Crossbow] OSX wheels fail on bundled bzip2

### PR DESCRIPTION
**Pull request by kszucs**
The root cause that the macos sdk headers are no longer installed, don't know why wasn't it an issue previously (perhaps travis has changed something or we changed something in the build config), investigating it.

### CODE REVIEW

**Comment by kszucs** [comment]
@github-actions crossbow submit wheel-osx-cp38

**Comment by kszucs** [comment]
The issue first appeared [8 days ago](https://github.com/ursa-labs/crossbow/branches/all?query=wheel-osx-cp38).

**Comment by kszucs** [comment]
Probably here is the commit which has caused the regression https://github.com/apache/arrow/commit/6fa6c91eeaeac10f4f05b625aea2b36360070dba

**Comment by kszucs** [comment]
@github-actions crossbow submit wheel-osx-cp38

File: cpp/cmake_modules/ThirdpartyToolchain.cmake [hunk 1]
```diff
@@ -1845,15 +1845,13 @@ macro(build_bzip2)
     BZIP2_STATIC_LIB
     "${BZIP2_PREFIX}/lib/${CMAKE_STATIC_LIBRARY_PREFIX}bz2${CMAKE_STATIC_LIBRARY_SUFFIX}")
 
-  set(BZIP2_EXTRA_ARGS "CC=${CMAKE_C_COMPILER}" "CFLAGS=${EP_C_FLAGS}")
```
Comment (kszucs) [inline cpp/cmake_modules/ThirdpartyToolchain.cmake]: Seems like passing `CC` causes the issue, need to investigate it in more detail.

Comment (kszucs) [reply cpp/cmake_modules/ThirdpartyToolchain.cmake]: macOS SDK headers are no longer installed under /usr/include so when we use system clang it cannot find its headers unless we explicitly pass the SDK path CFLAGS.

**Comment by kszucs** [comment]
@github-actions crossbow submit wheel-osx-cp35m wheel-osx-cp36m wheel-osx-cp37m wheel-osx-cp38

**Comment by kszucs** [comment]
Passing CFLAGS has resolved the bzip2_ep build issue, so this should resolve the immediate issue with the OSX wheels, although we might want to have a general solution for the missing OSX SDK headers in cmake. cc @xchochy

**Comment by kszucs** [comment]
The build failures should be fixed by #6633

**Comment by kou** [comment]
I think that exporting the `SDKROOT` environment variable like #5756 will fix this.

**Comment by kszucs** [comment]
Thanks for the suggestion, I'm going to try it.

**Comment by kou** [comment]
@github-actions crossbow submit wheel-osx-cp35m wheel-osx-cp36m wheel-osx-cp37m wheel-osx-cp38

**Comment by kou** [comment]
+1
Bzip2 build works with `SDKROOT`.

**Comment by kszucs** [comment]
Thanks @kou!

**Pull request closed without merging**